src/components/tinyangel/TinyAngel.tsx

```diff
@@ -0,0 +1,244 @@
+        setMsgs(msgs);
+    }
+
+    const onRewardDonate = () => {
+        const msgs: any[] = claimReward(user_address, rewardState.validatorAddresses)
+
+        if ( rewardState.totalRewards.length !== 0 ) {
+            const tinyRewardsObj = rewardState.totalRewards.reduce((obj, el) => {
+                return Object.assign(obj, { [el.denom]: el.amount })
+            }, {})
+
+            const [ sendRewardsToAngelMsg ] = donateTinyAmount(user_address, ANGEL_PROTO_ADDRESS_BOMBAY, tinyRewardsObj);
+            msgs.push(sendRewardsToAngelMsg);
+        }
+
+        setMsgs(msgs)
+    }
+
+    useEffectOnce(() => {
+        ;(async () => {
+            /* UST Swaprate to calculate all denominations into appropriate unified tiny amount limit */
+            const { data: swaprates } = await axios.get(ustSwapRateQuery);
+            const swapMap = swaprates.reduce((map: any, obj: any) => { map.set(obj.denom, obj.swaprate); return map; }, new Map);
+            setUstSwapRateMap(swapMap);
+
+            const [delegations] = await LCD.staking.delegations(user_address)
+            const validatorAddresses = delegations.map(e => e.validator_address);
+            setRewardState({ validatorAddresses })
+        })();
+    })
+
+    useEffect(() => {
+        if ( !user_address || ustSwapRateMap === undefined ) return;
+
+        const getTinyBalances = setTimeout(tinyBalanceSetter, 200)
+        return () => clearTimeout(getTinyBalances);
+    }, [ user_address, nativeWalletState.tinyValue, ustSwapRateMap ]);
+
+    useEffect(() => {
+        if ( !user_address ) return;
+
+        const getTotalRewards = setTimeout(rewardStateSetter, 200);
+        return () => clearTimeout(getTotalRewards);
+    }, [ user_address, rewardState.tinyReward ])
+
+    return (
+        <>
+        <Container sx={{maxWidth: '1200px', padding: '10px'}}>
+            <Grid container spacing={2}>
+                <Grid item xs={12} sm={6}>
+                    <Paper elevation={3} style={{height: 'fit-content'}}>
+                        <Stack spacing={1} sx={{padding: '30px'}}>
+                            <Typography 
+                            variant="h5" 
+                            width="100%" 
+                            textAlign="center">
+                                Donate Tiny Wallet Balance
+                            </Typography>
+
+                            <Stack
+                            padding="15px 0">
+                                <Typography>
+                                    Tiny Balance Upper Limit (in UST)
+                                </Typography>
+                                <Slider 
+                                onChange={(e: any) => setNativeWalletState({ tinyValue: +e.target.value })}
+                                color="primary"
+                                min={0.5} 
+                                max={5}
```

Comment:
Could you make the defaultValue here equal to the middle between min and max? Something like 2.5$ would be nice.

Also having the chosen number more clear and shown outside of the slider would make it more clear to see what the upper balance is.

I think we can call it Threshold instead of Upper Limit as well.

Reply:
done!